src/types/index.ts

```diff
@@ -34,3 +37,20 @@ export interface NuxtSecurityRouteRules {
   allowedMethodsRestricter?: AllowedHTTPMethods | false;
   nonce?: boolean;
 }
+
+export interface NuxtSecurityEventContext {
+  headers: Record<string, string | string[] | number | false> | null
+}
+
+declare module 'h3' {
+  interface H3EventContext {
+      security: NuxtSecurityEventContext
+  }
+}
+
+declare module 'nitropack' {
+  interface NitroRuntimeHooks {
+    'nuxt-security:headers': (route: string, headers: SecurityHeaders) => void
```

Comment:
I haven't dug into how the module works but this type of augmentation generally doesn't work when users install the module.

You need to use the runtime-generated types for this sort of thing usually.

Also I'd type it as `Promise<void> | void` so users can use promises.

Reply:
Personally I'd use a single context object instead of two separate params and I'd use a H3Event over the `route`.

Not required though

Reply:
yes, we probably need to add a reference in a `d.ts` file. I'll verify if this works later !

Reply:
would you prefer something like this?

```ts
'nuxt-security:headers': ({
  route: string,
  headers: SecurityHeaders
}) => void
```

Reply:
yes, it just makes it easier if you want to add other things into the hook context later